examples/appSpec/zomes/three/capabilities/main/code/index.ts

```diff
@@ -38,6 +41,54 @@ export function test_get(encoded_allocation: u32): u32 {
 }
 
 
+/*----------  test_decode_params boilerplate  ----------*/
+
+
+var valueString: string = "";
+
+class ParameterHandler extends Handler {
+  onString(value: string): void {
+    if(this.currentKey == "keyString") {
+        valueString = value;
+    }
+  }
+}
+
+export function test_decode_params(encoded_allocation: u32): u32 {
+   let jsonString: string = deserialize(encoded_allocation);   
+
+    let handler = new ParameterHandler()
+    parseString<ParameterHandler>(jsonString, handler)
```

Comment:
I saw in the commit history for assemblyscript that there is actually support for Object literals... did you know? Could parseString and giving it the handler actually return an object literal I wonder?

Reply:
Interesting. I actually wanted the handler to be an object literal rather than a subclass so it wouldn't have to be allocated on the heap but I couldn't get it to work for some reason.

Reply:
Oh yeah because there are no closures I couldn't write callbacks that modify variables from the scope they are defined in. So the handler still needs to be a class for now so it can keep track of the variables it is supposed to be decoding as properties.